### PR TITLE
Back Button

### LINKED ISSUE #68: Back Button with Routing
## Description

It is a button :) When clicked it should go back to the previous page (from company details page to search page) 

image

## Learning Objectives

- NextJs 
- Tailwind CSS

## Todo

These points are a rough guideline. Please feel free to discuss with others on the team about the best way to design this component!

- [ ] Create the button component 
  - when clicked it should go back to the previous page (from company details page to search page) 
  - check out NextJs useRouter
- [ ] Put it in company details page

## Useful Links

- [NextJs useRouter](https://nextjs.org/docs/api-reference/next/router)
- [Figma Designs](https://www.figma.com/file/PFCGAeZzwLEyzNaovh6Zfq/Toriis?node-id=3-1346&t=NARk7sp9fW2Lg70J-0)

### PR DESCRIPTION
**Pull request by eeshabarua**
## Summary

It is a button : ) it goes back 

Closes #68 

## Changes

- Create button component
- Implement routing
- Implement styling

## Screenshots

image

### CODE REVIEW

File: src/components/Buttons/BackButton.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import type { FC } from 'react'
+import { useRouter } from 'next/router'
+import { ArrowBack } from '@material-ui/icons'
```
Comment (daniel-moon32) [inline src/components/Buttons/BackButton.tsx:3]: I think we're using hero-icons, but @zorazrr can correct me if I'm wrong

File: src/components/Buttons/BackButton.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  const handleClick = () => {
+    router.back()
+  }
+
+  return (
+    <button
+      onClick={handleClick}
+      className={clsx(
+        clsx(
+          'justify-centerfont-klima flex w-20 place-content-center items-center text-[18px]',
```
Comment (daniel-moon32) [inline src/components/Buttons/BackButton.tsx:18]: I think generally we want to stray away from using a constant text size and width

File: src/components/Buttons/BackButton.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+          'shadow-gray shadow-lg',
+          'rounded-3xl border-[1px] border-black py-1 px-3',
+        ),
+      )}
+    >
+      <ArrowBack style={{ fontSize: 20 }} />
+      <p className="font-[400]"> Back</p>
```
Comment (daniel-moon32) [inline src/components/Buttons/BackButton.tsx:25]: Hmm shouldn't we use text-whateversize? instead of this?

**Review by daniel-moon32 — COMMENTED** [review]
Couple of minor styling issues, but looks good! Generally I would suggest looking at the build failure logs beforehand to double check any issues that you might be seeing.

File: src/components/Buttons/BackButton.tsx (hunk 1 above)
Comment (zorazrr) [reply src/components/Buttons/BackButton.tsx:3]: Right, we are using [hero icons](https://heroicons.com/)^^

File: src/pages/fossil-fuel/index.tsx [hunk 4]
```diff
@@ -1,7 +1,13 @@
 import type { FC } from 'react'
 
+import BackButton from '../../components/Buttons/BackButton'
+
 const Home: FC = () => {
-  return <div>Fossil Fuel Page</div>
+  return (
+    <>
+      <div>Fossil Fuel Page</div>
```
Comment (zorazrr) [inline src/pages/fossil-fuel/index.tsx:8]: Let's add the back button to company details page (should also merge main into your branch for this) !

File: src/components/Buttons/BackButton.tsx
Comment (zorazrr) [inline src/components/Buttons/BackButton.tsx:1]: Remember to export it in the [components master sheet](https://github.com/toriis-portal/toriis/blob/main/src/components/index.ts) whenever a new component is created!

File: src/components/Buttons/BackButton.tsx (hunk 2 above)
Comment (zorazrr) [inline src/components/Buttons/BackButton.tsx:18]:
```suggestion
          'justify-center font-klima flex w-20 place-content-center items-center text-[18px]',
```

**Review by zorazrr — CHANGES_REQUESTED** [review]
Some small things to change!

File: src/components/Buttons/BackButton.tsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+      className={clsx(
+        clsx(
+          'flex w-[5rem] place-content-center items-center justify-center font-klima text-[1.1rem]',
+          'shadow-gray shadow-lg',
+          'rounded-3xl border-[0.05rem] border-black py-1 ',
+        ),
+      )}
+    >
+      <ArrowSmallLeftIcon className="mr-1 inline w-[1rem] stroke-current stroke-1" />
+      <p className="font-[400]"> Back</p>
```
Comment (zorazrr) [inline src/components/Buttons/BackButton.tsx:25]:
```suggestion
      <p className="font-normal"> Back</p>
```

**Review by zorazrr — APPROVED** [review]
Great work!

**Pull request merged**